src/services/fungibleTokens.js

```diff
@@ -0,0 +1,90 @@
+import * as Sentry from '@sentry/browser';
+import BN from 'bn.js';
+import * as nearApiJs from 'near-api-js';
+
+import sendJson from '../tmp_fetch_send_json';
+import { wallet } from '../utils/wallet';
+import { ACCOUNT_HELPER_URL } from '../utils/wallet';
+
+const FT_MINIMUM_STORAGE_BALANCE = '1250000000000000000000';
+const FT_STORAGE_DEPOSIT_GAS = '30000000000000';
+const FT_TRANSFER_GAS = '30000000000000';
+const FT_TRANSFER_DEPOSIT = '1';
+
+const {
+    transactions: {
+        functionCall
+    }
+} = nearApiJs;
+
+class FungibleTokens {
+    constructor() {
+        this.account = wallet.getAccountBasic();
+    }
+
+    async isStorageBalanceAvailable(contractName, accountId) {
+        return new BN((await this.checkStorageBalance(contractName, accountId)).total).gte(new BN(FT_MINIMUM_STORAGE_BALANCE));
+    }
+
+    async checkStorageBalance(contractName, accountId) {
+        return await this.account.viewFunction(contractName, 'storage_balance_of', { account_id: accountId }).catch(logError);
+    }
+
+    async transfer(contractName, amount, memo, receiver) {
```

Comment:
Would be nice to make this take a single object as an argument...e.g.
```javascript
async transfer({ contractMetadata: { contractName, decimal}, amount, memo, receiver }) { ... }
```

It makes it easier to grok at the callsites that use the method, aids in composability, and makes changes less likely to be awkward in the future (for example, if we made `memo` optional).

Reply:
Are you sure about that, I mean it's easy to change but we don't use this approach in the project?

Reply:
Sorry about the delayed response, I just saw this comment(Github hid it!) :)

Re: am I sure, yes, named arguments are a Good Thing:tm: in a general way for readability, at the cost of a small amount of verbosity.

As an example, let us look at what it will look like when someone _calls_ this function:

```javascript
transferTo('accountId1', '15', 'accountId2', 'accountId3');
```

Without finding the implementation of `transferTo()` and reading it, I will have no idea what each of those arguments are. Is the receiver the first argument? Or is that the contract? The memo in this case is another NEAR account (for whatever reason) -- is that the receiver account? etc.

Imagine that `memo` becomes optional; this can present real problems that I have seen cause issues in the past:
```javascript
transferTo('accountId1', '15', null, 'accountId3');
```

Since it is now entirely likely that someone will actually pass `null` for the `memo`, we should probably add handling for `memo` being passed as `null` so we don't actually end up passing `null` to the call transfer call.  If we do, it will likely be automatically stringified into the string form of `null` which is actually a valid value for `memo`!
Javascript is tricky; it's in our long-term best interests to avoid needing this type of pattern; we can write simpler code if there is not an optional argument somewhere in the signature that is not the last argument.

Compare what it looks like to call the _same function_  if we use an object for the arguments:
```javascript
transferTo({
  contractMetadata: {
    contractName: 'accountId1'
  },
  amount: '10',
  memo: 'accountId2',
  receiver: 'accountId3'
});

// If `memo` becomes optional, _people can simply not provide it_ , and there is no incentive to pass `null` as a placeholder to maintain the same arity of the function.
transferTo({
  contractMetadata: {
    contractName: 'accountId1'
  },
  amount: '10',
  receiver: 'accountId3'
});
```

Even better, we can name `amount` something more explicit and it will be even clearer at the callsite -- e.g.

```javascript
transferTo({ 
  contractMetadata: {
    contractName: 'accountId1'
    ...
  },
  amountInYoctoNear: '10',
  receiver: 'accountId3'
});
```

Notice that the callsite where the function is _used_ is explicit about what you are doing; one can read the function call and immediately understand what each argument is doing without needing to read the implementation/full function definition of `transferTo()`.  This makes _reading the code_ a lot easier, even though it doesn't make any different to the computer, especially when you are reading code that you are unfamiliar with and trying to get an idea of what is happening without getting into the detail of exactly what each function's signature is.

It's also extremely nice for composability; order of arguments _is generally not important_; it only _becomes_ important if we have positional arguments.  If you are calling two or three functions in a row that take the same types as arguments, it is much less cognitive overhead if each of those 3 functions doesn't have the types in a different order.  Usually with named arguments, all 3 functions can have _the same name for an argument of a particular type_; you never have to shuffle the order of things because functions have slightly different orders of arguments.  Also as we move to typescript, re-using and composing types is a lot easier if we use objects to define function arguments; positional arguments tend to make it difficult to re-use types, especially when some functions take subsets of types that their calling functions take as arguments; you end up with a lot of very explicit type signatures for functions rather than using a few domain-driven types and passing subsets of those types around internally (see https://www.typescriptlang.org/docs/handbook/utility-types.html).

I understand we often have used positional arguments exclusively, even for functions with a large number of arguments and for functions with optional arguments in the past, but I want this to change, _especially_ for functions with multiple arguments, optional arguments, or arguments that we can conceivably imagine becoming optional.

This also aligns with an initiative happening in `near-api-js` to replace many function signatures that use  positional arguments to take an object as an argument instead.

cc: @Patrick1904 as this is relevant on a macro level to the codebase as a whole, and to the entire JS ecosystem at NEAR

I recommend spending a little time and watching this video and internalizing the ideas presented there, as it is a really good overview about how we should reason about cohesion and coupling in software:
https://www.youtube.com/watch?v=Ip2o4vbAK3s

What we're discussing here now is _removing connascence of order_.
We only _want_ _connascence of order_ when the _order itself is important_ in some way.  If we add connascence of order to things where the order is incidental or unimportant, all we are doing is adding additional coupling that we need to work around from then on.  Connascence of name is weaker than connascence of order and it is easier to reason about and keep consistent :)

Reply:
Got it, I wasn't sure if you would like to use this approach in the whole project. Done.